### PR TITLE
Check if we need to link against stdc++fs

### PR DESCRIPTION
**Pull request by alexsavulescu**
* https://en.cppreference.com/w/cpp/filesystem#Notes
* Revert "CLI11 & GCC8 combo have special consideration  #839"
  * This reverts commit 3a274647a7cd96fc6f1538096a69aa6b01fcdfad.

### CODE REVIEW

File: coreneuron/CMakeLists.txt [hunk 1]
```diff
@@ -260,6 +260,12 @@ target_include_directories(coreneuron SYSTEM
 target_include_directories(coreneuron SYSTEM
                            PRIVATE ${CORENEURON_PROJECT_SOURCE_DIR}/external/CLI11/include)
 
+# See: https://en.cppreference.com/w/cpp/filesystem#Notes
```
[the remark this answers is not in the document]
Comment (alexsavulescu) [reply coreneuron/CMakeLists.txt:263]: Just tested on my system wiht GCC7 and it works
```
(venv) savulesc@bbd-cjngk03:~/Workspace/nmodl/build/nrn/build$ gcc-7 --version
gcc-7 (Ubuntu 7.5.0-6ubuntu2) 7.5.0
Copyright (C) 2017 Free Software Foundation, Inc.
This is free software; see the source for copying conditions.  There is NO
warranty; not even for MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
```

[the remark this answers is not in the document]
Comment (olupton) [reply coreneuron/CMakeLists.txt:263]: OK. Looks like it's been in GCC for a while judging from: https://gitlab.kitware.com/cmake/cmake/-/issues/17834.

Not sure about the Clang part. IIUC this is only correct if Clang is using `libstdc++` from GCC8 or older, not a newer `libstdc++` (typically recent Linux) or `libc++` (typically macOS). I don't know a nice way of detecting when which link library is required...

[the remark this answers is not in the document]
Comment (alexsavulescu) [reply coreneuron/CMakeLists.txt:263]: > I don't know a nice way of detecting when which link library is required...

I guess we can drop that part or leave it as is and see if it comes back to bite us ..

[the remark this answers is not in the document]
Comment (olupton) [reply coreneuron/CMakeLists.txt:263]: I would suggest dropping the Clang part, assuming that our CI lets us...

**Review by olupton — APPROVED** [review]
LGTM (well, 🤢, but...) 🚀

File: coreneuron/CMakeLists.txt [hunk 2]
```diff
@@ -260,6 +260,11 @@ target_include_directories(coreneuron SYSTEM
...
+if(CMAKE_CXX_COMPILER_IS_GCC AND CMAKE_CXX_COMPILER_VERSION VERSION_LESS 9.1)
+  target_link_libraries(coreneuron stdc++fs)
+endif()
```
Comment (pramodk) [inline coreneuron/CMakeLists.txt:266]: note: as this is a compile-time decision here, this is not sufficient if we consider the use case of wheels:
* wheel is built with gcc > 9.1 and hence `stdc++fs` is not linked
* user is now using neuron wheel with gcc < 9.1. this results into link error

**Pull request merged**